packages/@tailwindcss-upgrade/src/template/codemods/migrate-important.ts

```diff
@@ -0,0 +1,23 @@
+import type { Candidate } from '../../../../tailwindcss/src/candidate'
+
+// In v3 the important modifier `!` sits in front of the utility itself, not
+// before any of the variants. In v4, we want it to be at the end of the utility
+// so that it's always in the same location regardless of whether you used
+// variants or not.
+//
+// So this:
+//
+//   !flex md:!block
+//
+// Should turn into:
+//
+//   flex! md:block!
+export function migrateImportant(candidate: Candidate): Candidate | null {
+  if (candidate.important) {
```

Comment:
To prevent migrating `md:block!`, maybe we can check if the `!` is already in the correct position? (assuming we track the raw candidate we parsed in the AST)
```suggestion
  if (candidate.important && candidate.raw[candidate.raw.length - 1] !== '!') {
```

This would then result in fewer migrations that aren't necessary, but also in cases like this: 
- In: `min-[calc(1000px+12em)]:flex!`
- Out: `min-[calc(1000px_+_12em)]:flex!`

Reply:
This makes sense yepp